python/felis/cli.py

```diff
@@ -71,26 +71,58 @@ def cli(log_level: str, log_file: str | None) -> None:
         logging.basicConfig(level=log_level)
 
 
-@cli.command("create-all")
-@click.option("--engine-url", envvar="ENGINE_URL", help="SQLAlchemy Engine URL")
-@click.option("--schema-name", help="Alternate Schema Name for Felis File")
-@click.option("--dry-run", is_flag=True, help="Dry Run Only. Prints out the DDL that would be executed")
+@cli.command("create")
+@click.option("--engine-url", envvar="ENGINE_URL", help="SQLAlchemy Engine URL", default="sqlite://")
+@click.option("--schema-name", help="Alternate schema name to override Felis file")
+@click.option(
+    "--create-if-not-exists", is_flag=True, help="Create the schema in the database if it does not exist"
+)
+@click.option("--drop-if-exists", is_flag=True, help="Drop schema if it already exists in the database")
+@click.option("--echo", is_flag=True, help="Echo database commands as they are executed")
+@click.option("--dry-run", is_flag=True, help="Dry run only to print out commands instead of executing")
+@click.option(
+    "--output-file",
+    "-o",
+    type=click.File(mode="w"),
+    help="Write SQL commands to a file instead of executing",
+    required=False,
+)
 @click.argument("file", type=click.File())
-def create_all(engine_url: str, schema_name: str, dry_run: bool, file: io.TextIOBase) -> None:
-    """Create schema objects from the Felis FILE."""
-    schema_obj = yaml.load(file, Loader=yaml.SafeLoader)
-    visitor = SQLVisitor(schema_name=schema_name)
-    schema = visitor.visit_schema(schema_obj)
-
-    metadata = schema.metadata
-
+def create(
+    engine_url: str,
+    schema_name: str | None,
+    create_if_not_exists: bool,
+    drop_if_exists: bool,
+    echo: bool,
+    dry_run: bool,
+    output_file: io.TextIOBase | None,
+    file: io.TextIOBase,
+) -> None:
+    """Create database objects from the Felis file."""
+    schema = Schema.from_yaml_file(file)
+    url_obj = make_url(engine_url)
+    metadata = SchemaMetaData(
+        schema, schema_name, no_metadata_schema=True if url_obj.drivername == "sqlite" else False
+    )
+    logger.debug(f"Created metadata with schema name: {metadata.schema}")
     engine: Engine | MockConnection
-    if not dry_run:
-        engine = create_engine(engine_url)
+    if not dry_run and not output_file:
+        engine = create_engine(engine_url, echo=echo)
     else:
-        _insert_dump = InsertDump()
-        engine = create_mock_engine(make_url(engine_url), executor=_insert_dump.dump)
-        _insert_dump.dialect = engine.dialect
+        if dry_run:
+            logger.info("Dry run will be executed")
+        dumper = InsertDump(output_file)
+        engine = create_mock_engine(make_url(engine_url), executor=dumper.dump)
+        dumper.dialect = engine.dialect
+        if output_file:
+            logger.info("Writing SQL output to: " + getattr(output_file, "name", ""))
```

Comment:
```suggestion
            logger.info("Writing SQL output to: %s", output_file.name)
```
Although I don't really know why the getattr is needed since `output_file` has already been checked (does `mypy` want you to write `if output_file is not None`?

Reply:
mypy complains when I try to directly access the `name` attribute:

```
python/felis/cli.py:119: error: "TextIOBase" has no attribute "name"  [attr-defined]
```

Reply:
You are telling it it is a `TextIOBase` and then requiring that `name` exists. What class are you really expecting it to be?

Reply:
> You are telling it it is a `TextIOBase` and then requiring that `name` exists. What class are you really expecting it to be?

I'm not sure what IO class to use here which has a name attribute that the type system can actually see. (It works fine with duck typing but `name` must be in a child class that it can't see.)

Reply:
Does 

```python
from typing import IO
...
output_file: IO[str]
```
work?

Reply:
I changed this to `typing.IO` which has a visible `name` attribute.